### PR TITLE
Add loading OpenAPI spec from JSON string

### PR DESCRIPTION
**Pull request by nadheesh**
## Purpose
Resolves https://github.com/wso2-enterprise/choreo/issues/22973

### CODE REVIEW

**Review by JayaniH — APPROVED** [review]
LGTM

File: ballerina/openapi_utils.bal [hunk 1]
```diff
@@ -35,13 +36,33 @@ public type AdditionInfoFlags record {|
 
 # Extracts the Http tools from the given OpenAPI specification file.
 #
-# + filePath - Path to the OpenAPI specification file
+# + filePath - Path to the OpenAPI specification file (should be JSON or YAML)
 # + additionInfoFlags - Flags to extract additional information from the OpenAPI specification
 # + return - A record with the list of extracted tools and the service URL (if available)
-public function extractToolsFromOpenApiSpec(string filePath, *AdditionInfoFlags additionInfoFlags) returns HttpApiSpecification & readonly|error {
-    OpenApiSpec openApiSpec = check parseOpenApiSpec(filePath);
+public function extractToolsFromOpenApiSpecFile(string filePath, *AdditionInfoFlags additionInfoFlags) returns HttpApiSpecification & readonly|error {
```
Comment (anupama-pathirage) [inline ballerina/openapi_utils.bal:42]: Break the line by 120 max chars.

File: ballerina/openapi_utils.bal [hunk 2]
```diff
@@ -35,13 +36,33 @@ public type AdditionInfoFlags record {|
...
+    map<json> openApiSpec;
+    if filePath.endsWith(".yaml") || filePath.endsWith(".yml") {
+        openApiSpec = check yaml:readFile(filePath).ensureType();
+    }
+    else if filePath.endsWith(".json") {
+        openApiSpec = check io:fileReadJson(filePath).ensureType();
+    }
+    else {
+        return error("Unsupported file type. Supported file types are .json or .yaml");
```
Comment (anupama-pathirage) [inline ballerina/openapi_utils.bal:51]:
```suggestion
        return error("Unsupported file type. Supported file types are .json or .yaml");
```
```suggestion
        return error("Unsupported file type. Supported file types are .json, .yaml or .yml");
```

File: ballerina/openapi_utils.bal (hunk 1 above)
Comment (nadheesh) [reply ballerina/openapi_utils.bal]: fixed

File: ballerina/openapi_utils.bal [hunk 3]
```diff
@@ -35,13 +36,33 @@ public type AdditionInfoFlags record {|
...
+    }
+    return extractToolsFromOpenApiJsonSpec(openApiSpec, additionInfoFlags);
+}
+
+# Extracts the Http tools from the given OpenAPI specification as a JSON 
+#
+# + openApiSpec - A valid OpenAPI specification in JSON format
+# + additionInfoFlags - Flags to extract additional information from the OpenAPI specification
+# + return - A record with the list of extracted tools and the service URL (if available)
+public function extractToolsFromOpenApiJsonSpec(map<json> openApiSpec, *AdditionInfoFlags additionInfoFlags) returns HttpApiSpecification & readonly|error {
```
[the remark this answers is not in the document]
Comment (nadheesh) [reply ballerina/openapi_utils.bal]: fixed

**Pull request merged**